lib/date_time/date_time_ntp.c

```diff
@@ -15,69 +15,43 @@
 
 LOG_MODULE_DECLARE(date_time, CONFIG_DATE_TIME_LOG_LEVEL);
 
-#define UIO_IP      "ntp.uio.no"
-#define GOOGLE_IP_1 "time1.google.com"
-#define GOOGLE_IP_2 "time2.google.com"
-#define GOOGLE_IP_3 "time3.google.com"
-#define GOOGLE_IP_4 "time4.google.com"
-
-#define NTP_PORT "123"
-
-struct ntp_servers {
-	const char *server_str;
-	struct sockaddr addr;
-	socklen_t addrlen;
-};
+#define UIO_NTP    "ntp.uio.no"
+#define GOOGLE_NTP "time.google.com"
+
+#define NTP_PORT 123
 
-struct ntp_servers servers[] = {
-	{.server_str = UIO_IP},
-	{.server_str = GOOGLE_IP_1},
-	{.server_str = GOOGLE_IP_2},
-	{.server_str = GOOGLE_IP_3},
-	{.server_str = GOOGLE_IP_4}
+static const char * const servers[] = {
+	UIO_NTP,
+	GOOGLE_NTP
 };
 
 static struct sntp_time sntp_time;
 
-static int sntp_time_request(struct ntp_servers *server, uint32_t timeout,
-			     struct sntp_time *time)
+static int sntp_time_request(const char *server, uint32_t timeout, struct sntp_time *time)
 {
 	int err;
-	static struct addrinfo hints;
 	struct addrinfo *addrinfo;
 	struct sntp_ctx sntp_ctx;
 
-	if (server->addrlen == 0) {
-		if (IS_ENABLED(CONFIG_DATE_TIME_IPV6)) {
-			hints.ai_family = AF_INET6;
-		} else {
-			hints.ai_family = AF_INET;
-		}
-		hints.ai_socktype = SOCK_DGRAM;
-		hints.ai_protocol = 0;
+	struct addrinfo hints = {
+		.ai_flags = AI_NUMERICSERV,
+		.ai_socktype = SOCK_STREAM
+	};
 
-		err = getaddrinfo(server->server_str, NTP_PORT, &hints,
-				  &addrinfo);
-		if (err) {
-			LOG_WRN("getaddrinfo, error: %d", err);
-			return err;
-		}
-
-		if (addrinfo->ai_addrlen > sizeof(server->addr)) {
-			LOG_WRN("getaddrinfo, addrlen: %d > %d",
-				addrinfo->ai_addrlen, sizeof(server->addr));
-			freeaddrinfo(addrinfo);
-			return -ENOMEM;
-		}
-
-		memcpy(&server->addr, addrinfo->ai_addr, addrinfo->ai_addrlen);
-		server->addrlen = addrinfo->ai_addrlen;
-		freeaddrinfo(addrinfo);
+	if (IS_ENABLED(CONFIG_DATE_TIME_IPV6)) {
+		/* Allow both IPv6 and IPv4. */
+		hints.ai_family = AF_UNSPEC;
```

Comment:
Another option would be to remove the `CONFIG_DATE_TIME_IPV6` option. I think it should be enough that we always use `AF_UNSPEC`, which uses IPv6 when available and otherwise IPv4. I can't think of a reason to force IPv4 usage when IPv6 could be used instead.

Reply:
Yep, that's my proposal too.

Reply:
`CONFIG_DATE_TIME_IPV6` has now been removed.